controllers/scenario/scenario_adapter.go

```diff
@@ -0,0 +1,164 @@
+
+		patch := client.MergeFrom(a.scenario.DeepCopy())
+		SetScenarioIntegrationStatusAsInvalid(a.scenario, "Failed to get application for scenario.")
+		err := a.client.Status().Patch(a.context, a.scenario, patch)
+		if err != nil {
+			a.logger.Error(err, "Failed to update Scenario",
+				"Scenario.Name", a.scenario.Name)
+			return reconciler.RequeueWithError(err)
+		}
+		return reconciler.ContinueProcessing()
+	}
+	// Checks if scenario has ownerReference assigned to it
+	if a.scenario.OwnerReferences == nil {
+		patch := client.MergeFrom(a.scenario.DeepCopy())
+		err := ctrl.SetControllerReference(a.application, a.scenario, a.client.Scheme())
+		if err != nil {
+			a.logger.Error(err, "Error setting owner reference.",
+				"Application.Name", a.application.Name,
+				"Application.Namespace", a.application.Namespace)
+			patch := client.MergeFrom(a.scenario.DeepCopy())
+			SetScenarioIntegrationStatusAsInvalid(a.scenario, "Failed to get application for scenario.")
+			err = a.client.Status().Patch(a.context, a.scenario, patch)
+			if err != nil {
+				a.logger.Error(err, "Failed to update Scenario",
+					"Scenario.Name", a.scenario.Name)
+				return reconciler.RequeueWithError(err)
+			}
+			return reconciler.ContinueProcessing()
+		}
+		err = a.client.Patch(a.context, a.scenario, patch)
+		if err != nil {
+			a.logger.Error(err, "Failed to update Scenario",
+				"Scenario.Name", a.scenario.Name)
+			return reconciler.RequeueWithError(err)
+		}
+		return reconciler.ContinueProcessing()
+	}
+	// Checks if scenario has environment defined
+	if reflect.ValueOf(a.scenario.Spec.Environment).IsZero() {
+		a.logger.Info("IntegrationTestScenario has no environment defined",
+			"Scenario.Name:", a.scenario.Name,
+			"Scenario.Namespace", a.scenario.Namespace)
+		return reconciler.ContinueProcessing()
+	} else {
+		//Same as function getEnvironmentFromIntegrationTestScenario - this could be later changed to call only that function
+		ITSEnv := &applicationapiv1alpha1.Environment{}
+
+		err := a.client.Get(a.context, types.NamespacedName{
+			Namespace: a.application.Namespace,
+			Name:      a.scenario.Spec.Environment.Name,
+		}, ITSEnv)
+
+		if err != nil {
+			a.logger.Info("Environment doesn't exist in same namespace as IntegrationTestScenario.",
+				"Scenario.Name:", a.scenario.Name,
+				"Scenario.Namespace:", a.scenario.Namespace,
+				"Environment.Name:", a.scenario.Spec.Environment.Name)
+			patch := client.MergeFrom(a.scenario.DeepCopy())
+			SetScenarioIntegrationStatusAsInvalid(a.scenario, "Environment "+a.scenario.Spec.Environment.Name+" is located in different namespace than scenario.")
+			err = a.client.Status().Patch(a.context, a.scenario, patch)
+			if err != nil {
+				a.logger.Error(err, "Failed to update Scenario",
+					"Scenario.Name", a.scenario.Name)
+				return reconciler.RequeueWithError(err)
+			}
+			return reconciler.ContinueProcessing()
+		}
+
+	}
+	if reflect.ValueOf(a.scenario.Status).IsZero() {
```

Comment:
This condition only seems to kick-in when the IntegrationTestScenario status is not set. I think we want to check if the `IntegrationTestScenarioValid` condition is set to false instead. Here's an example of how we check the [Snapshot status condition.](https://github.com/redhat-appstudio/integration-service/blob/main/gitops/snapshot.go#L168)

Reply:
Turned out that valid ITS was not handled properly because of too many ` return reconciler.ContinueProcessing()` statements. It gets properly handled now.